taskgraph/Task.py

```diff
@@ -1209,9 +1209,12 @@ def is_precalculated(self):
                         mismatched_target_file_list.append(
                             "Path names don't match\n"
                             "cached: (%s)\nactual (%s)" % (path, actual_path))
-                    target_modified_time = os.path.getmtime(path)
-                    if not math.isclose(
-                            float(modified_time), target_modified_time):
+
+                    # Using nanosecond resolution for mtime (instead of the
+                    # usual float result of os.path.getmtime()) allows us to
+                    # precisely compare modification time.
+                    target_modified_time = os.stat(path).st_mtime_ns
+                    if not int(modified_time) == target_modified_time:

```

Comment:
possibly worth a note that this works because `st_mtime_ns` is always an integer?

Reply:
Also I would add to the readme or wherever makes sense that this may not work on some systems, which do not have nanosecond resolution timestamps

Reply:
I've added to the inline comment here in 6fc2bf2 and added a "Caveats" section to the README in 36acdd4.

The timestamp question is a fun rabbit hole!  Even a number of semi-modern filesystems have much poorer timestamp resolution than I expected.